internal/streamingnode/server/wal/interceptors/segment/inspector/impls.go

```diff
@@ -121,9 +124,16 @@ func (s *sealOperationInspectorImpl) background() {
 				return true
 			})
 		case <-mustSealTicker.C:
-			segmentBelongs := resource.Resource().SegmentAssignStatsManager().SealByTotalGrowingSegmentsSize()
+			threshold := paramtable.Get().DataCoordCfg.GrowingSegmentsMemSizeInMB.GetAsUint64() * 1024 * 1024
+			segmentBelongs := resource.Resource().SegmentAssignStatsManager().SealByTotalGrowingSegmentsSize(threshold)
+			if segmentBelongs == nil {
```

Comment:
Since SealByTotalGrowingSegmentsSize always returns a non-nil value, the check is invalid.